src/api-sdk/github/utils.ts

```diff
@@ -0,0 +1,85 @@
+import { ApiResponse, ApiErrorResponse } from '../response';
+
+export const extractBaseShaHelper = (response: ApiResponse<any> | ApiErrorResponse) => {
+    if ('resource' in response && response.resource && 'object' in response.resource && 'sha' in response.resource.object) {
+        return response.resource.object.sha;
+    } else {
+        return response;
```

Comment:
We should check if we have an error and throw it immediately instead of propagating it and return the `sha` otherwise

Reply:
Hey Mo, if we check we have an error first then return the `sha` we still need to check that the `response` contains `resource` due to our typing. 

Something like:

```
export const extractBaseShaHelper = (response: ApiResponse<any> | ApiErrorResponse) => {
    if ('errors' in response) {
        throw new Error(`API Error: ${JSON.stringify(response.errors)}`);
    } else if ('resource' in response) {
        return response.resource.object.sha;
    }
};
```

Is this fine with you or do you have any further suggestions?

Reply:
> ```
> export const extractBaseShaHelper = (response: ApiResponse<any> | ApiErrorResponse) => {
>     if ('errors' in response) {
>         throw new Error(`API Error: ${JSON.stringify(response.errors)}`);
>     } else if ('resource' in response) {
>         return response.resource.object.sha;
>     }
> };
> ```
> 
> Is this fine with you or do you have any further suggestions?

```suggestion
 export const extractBaseShaHelper = (response: ApiResponse<any> | ApiErrorResponse) => {
    if('resource' in response){
        return response.resource.sha;
    }
    throw new Error(`Error: ${JSON.stringify(response)}`);
}          
```